app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java

```diff
@@ -446,7 +446,7 @@ public void updateArtifactState(String groupId, String artifactId, ArtifactState
     /**
      * @see RegistryStorage#updateArtifactState(java.lang.String, java.lang.String, java.lang.String, io.apicurio.registry.types.ArtifactState)
      */
-    @Override
+    @Override @Transactional
```

Comment:
This change can affect performance, do you think it can significantly impact?

Reply:
I don't think so.  Especially since the state change endpoint is very rarely used.